pkg/clients/ec2/internetgateway.go

```diff
@@ -35,3 +48,53 @@ func IsInternetGatewayNotFoundErr(err error) bool {
 
 	return false
 }
+
+// IsInternetGatewayAlreadyAttached returns true if the error is because the item doesn't exist
+func IsInternetGatewayAlreadyAttached(err error) bool {
+	if awsErr, ok := err.(awserr.Error); ok {
+		if awsErr.Code() == InternetGatewayAlreadyAttached {
+			return true
+		}
+	}
+
+	return false
+}
+
+// GenerateIGObservation is used to produce v1beta1.InternetGatewayExternalStatus from
+// ec2.InternetGateway.
+func GenerateIGObservation(ig ec2.InternetGateway) v1beta1.InternetGatewayObservation {
+	attachments := make([]v1beta1.InternetGatewayAttachment, len(ig.Attachments))
+	for k, a := range ig.Attachments {
+		attachments[k] = v1beta1.InternetGatewayAttachment{
+			AttachmentStatus: string(a.State),
+			VPCID:            aws.StringValue(a.VpcId),
+		}
+	}
+
+	return v1beta1.InternetGatewayObservation{
+		InternetGatewayID: aws.StringValue(ig.InternetGatewayId),
+		Attachments:       attachments,
+		OwnerID:           aws.StringValue(ig.OwnerId),
+	}
+}
+
+// IsIgUpToDate checks whether there is a change in any of the modifiable fields.
+func IsIgUpToDate(p v1beta1.InternetGatewayParameters, ig ec2.InternetGateway) bool {
+	attachments := ig.Attachments
+	upToDate := false
```

Comment:
I think just returning `false` in any comparison statements below should be fine without needing separate `upToDate` variable.